oidc-exchange.py

```diff
@@ -0,0 +1,65 @@
+import os
+import sys
+
+import requests
+
+
+def die(msg: str):
+    print(f"OIDC exchange failure: {msg}", file=sys.stderr)
+    sys.exit(1)
+
+
+oidc_audience = os.getenv("INPUT_OIDC_AUDIENCE")
+
+oidc_token_url = os.getenv("ACTIONS_ID_TOKEN_REQUEST_URL")
+if oidc_token_url is None:
+    die("no ACTIONS_ID_TOKEN_REQUEST_URL in environment")
+
+oidc_bearer_token = os.getenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+if oidc_bearer_token is None:
+    die("no ACTIONS_ID_TOKEN_REQUEST_TOKEN in environment")
+
+oidc_resp = requests.get(
+    oidc_token_url,
+    params={"audience": oidc_audience},
+    headers={"Authorization": f"bearer {oidc_bearer_token}"},
+)
+
+if not oidc_resp.ok:
+    die(f"OIDC credential request error: {oidc_resp.status_code}")
+
+oidc_payload = oidc_resp.json()
+oidc_token = oidc_payload.get("value")
+if oidc_token is None:
+    # We emit the OIDC payload here, knowing that GitHub Actions will mask
+    # the token contents (if present).
+    die(f"malformed OIDC credential response: {oidc_payload}")
+
+# Now we can do the actual token exchange.
+repository_url = os.getenv("INPUTS_REPOSITORY_URL")
```

Comment:
I have no objection to this, but is there a benefit to it? It feels like it'd be slightly out of sync with everything else (since everything else comes from the env).

Reply:
Okay, I thought about passing everything in. But you're right, it's not the time. It'd be better structured if everything was in Python. Until that happens, it's probably not worth it...